TESTING.md

```diff
@@ -0,0 +1,59 @@
+
+The main difference between SimpleCov and ChefSpec coverage is that SimpleCov reports line coverage and ChefSpec shows custom resource touch coverage.
+
+### Style checking
+Style checking is done by `rubocop` for all the custom resource providers and pure ruby lib files, and `foodcritic` to analyze testing recipes and examples.
+
+### Functional tests
+Every resource must have an example for each action that shows how to use it. These examples can be used to do functional and OneView integration testing but they need to be executed manually using the Chef.
+
+## Implementing tests
+All the code must have associated tests, be it the already implemented or new submitted code, and this section covers what tests need to be implemented.
+
+### Unit tests
+The unit tests are required for each pure ruby lib methods and for each Chef custom resource action.
+
+#### Minimum requisites
+The necessary amount of testing is dictated by who is implementing and by who is accepting the code, however, there is a floor acceptance level that must be followed:
+  1. All the pure ruby lib code must have unit tests hitting each line of them, i.e., the line coverage for this kind of code must be 100%.
+  2. Each custom resource must have at least one unit test for each resource action.
+
+#### Lib file tests
+The pure ruby lib files are tested using common RSpec syntax, and they are located in `spec/unit/libraries`.
+
+#### Custom resource tests
+The custom resources are tested using the ChefSpec syntax, similar to RSpec, but they need some extra information to work:
+  1. The actual tests are located in `spec/unit/resource`-like folders. Each resource has a folder, and, inside these folders, there is one test file for each custom resource action (or use case, if necessary).
+  2. The fixtures are fake cookbooks bound to a hardware variant and API version that contain fixture recipes that uniquely executes one action for a specific resource. They are located in `spec/fixtures`.
+  3. Each test file has at least one corresponding fixture recipe. (They may have more than one if testing new features from newer API versions)
+  4. Each `it` clause must execute the fixture recipe, and that is why the code must create custom matchers to execute the Chef run and check if the correct action was triggered.
+
+#### Creating the custom matchers
+The custom matchers need to be created in the `library/matchers.rb` file. There you must add the action verb of your new action to the action list if it does not exists, and add the full action name to your resource in the matrix.
```

Comment:
The custom matchers need to be created in the `library/matchers.rb` file. There you must add the action verb of your new action to the action list if it does not exist ~s~, and add the full action name to your resource in the matrix.